proposals/lip-0045.md

```diff
@@ -1313,6 +1317,52 @@ def error(ccm: CCM, errorStatus: uint32) -> None:
     )
 ```
 
+#### isChannelActive
+
+The `isChannelActive` function checks that a chain (or the mainchain if the chain account does not exist) has status `CHAIN_STATUS_ACTIVE`. On the mainchain, it also checks the liveness condition for the chain.
+
+```python
+def isChannelActive(chainID: ChainID) -> bool:
+    # We do not consider the channel active if it is the own chain.
+    if chainID == ownChainAccount.chainID:
+        return False
+
+    # Processing on the mainchain.
+    if ownChainAccount.chainID == getMainchainID():
+        # Account has to exist on the mainchain.
+        if chainAccount(chainID) does not exist:
+            return False
+        
+        # Check if chain status is active.
+        if not chainAccount(chainID).status == CHAIN_STATUS_ACTIVE:
+            return False
+        
+        # Check liveness condition for active chains.
+        timestamp = timestamp of the block where this state transition is processed
+        if timestamp - chainAccount(chainID).lastCertificate.timestamp > LIVENESS_LIMIT:
+            return False
+
+        return True
+        
+    # Processing on a sidechain.
+    else:
+        # Account may not exist on a sidechain.
```

Comment:
I am a bit confused, by comparing this to the `isLive` function. 

In case we are in a sidechain, and `chainID` is another sidechain with no `chainAccount(chainID)`, then `isLive` returns `False` in case there exists a `terminatedStateAccount(chainID)`.

It seems to me that in this case, `isChannelActive()` returns True (assuming mainchain is active), which does not seem correct to me.

Reply:
if there is terminatedStateAccount(chainID) than status is CHAIN_STATUS_TERMINATED